common/lib/xmodule/xmodule/video_module/video_module.py

```diff
@@ -934,28 +940,53 @@ def _parse_video_xml(cls, xml, id_generator=None):
 
         return field_data
 
-    def import_video_info_into_val(self, xml, course_id):
+    def import_video_info_into_val(self, xml, resource_fs, course_id):
         """
         Import parsed video info from `xml` into edxval.
 
         Arguments:
-            xml (lxml object): xml representation of video to be imported
+            xml (lxml object): xml representation of video to be imported.
+            resource_fs (OSFS): Import file system.
             course_id (str): course id
         """
-        edx_video_id = ''
-        if self.edx_video_id is not None:
-            edx_video_id = self.edx_video_id.strip()
+        edx_video_id = clean_video_id(self.edx_video_id)
 
+        # Create video_asset is not already present.
         video_asset_elem = xml.find('video_asset')
-        if edxval_api and video_asset_elem is not None:
-            # Always pass the edx_video_id, Whether the video is internal or external
-            # In case of external, we only need to import transcripts and for that
-            # purpose video id is already present in the xml
-            edxval_api.import_from_xml(
+        if video_asset_elem is None:
+            video_asset_elem = etree.Element('video_asset')
+            video_asset_elem.set('duration', '0.0')
+            video_asset_elem.set('client_video_id', u'External Video')
+
+        # This will be a dict containing the list of names of the external transcripts.
+        # Example:
+        # {
+        #     'en': ['The_Flash.srt'],
+        #     'es': ['Green_Arrow.srt', 'Harry_Potter.srt']
+        # }
+        external_transcripts = defaultdict(list)
+
+        # Add trancript from self.sub and self.youtube_id_1_0 fields.
+        external_transcripts[self.transcript_language] = [
```

Comment:
what if `self.transcript_language` isn't `en`?

Reply:
@mushtaqak we need to do testing around this. BTW what is the current behaviour?